packages/runtime/src/helpers/files.ts

```diff
@@ -333,8 +333,13 @@ const getServerFile = (root: string, includeBase = true) => {
   return findModuleFromBase({ candidates, paths: [root] })
 }
 
+// force manual revalidation so the ODB handler always receives fresh content
+// ensure fallback behaviour is bypassed for pre-rendered pages
+// ensure ISR 404 pages send the correct SWR cache headers
 const baseServerReplacements: Array<[string, string]> = [
-  [`let ssgCacheKey = `, `let ssgCacheKey = process.env._BYPASS_SSG || `],
+  [`checkIsManualRevalidate(req, this.renderOpts.previewProps)`, `checkIsManualRevalidate({ headers: null }, null)`],
+  [`isManualRevalidate && (fallbackMode !== false || hadCache)`, `isManualRevalidate && hadCache`],
+  [`private: isPreviewMode || is404Page && cachedData`, `private: isPreviewMode && cachedData`],
```

Comment:
These are the new patches for forcing manual revalidation. It's also possible to force this by sending an `x-prerender-revalidate` request header in the handler, but that involves pulling in the `prerender-manifest.json` to get a token and I thought this might actually be more succinct